src/librbd/operation/SnapshotRemoveRequest.cc

```diff
@@ -137,7 +137,42 @@ void SnapshotRemoveRequest<I>::handle_get_snap(int r) {
 
     auto it = m_out_bl.cbegin();
     r = cls_client::snapshot_get_finish(&it, &snap_info);
-    if (r == 0) {
+    if (r == 0 && snap_info.child_count > 0) {
+      cls::rbd::ChildImageSpecs child_images;
+      int r2 = cls_client::children_list(&image_ctx.md_ctx,
+                                         image_ctx.header_oid,
+                                         m_snap_id, &child_images);
+      if (r2 < 0 && r2 != -ENOENT && r2 != -EOPNOTSUPP) {
```

Comment:
Nit: if `child_count` is greater then zero, you shouldn't have to worry about `-EOPNOTSUPP`